packages/backend/src/core/discovery/DiscoveryEngine.ts

```diff
@@ -26,7 +34,16 @@ export class DiscoveryEngine {
       blockNumber,
     )
 
-    const discovered = await discover(discoveryProvider, config, this.logger)
+    const discovered = await wrapAndMeasure(
+      async () => await discover(discoveryProvider, config, this.logger),
+      {
+        histogram: discoveryHistogram,
+        labels: {
+          project: config.name,
+        },
+      },
+    )()
```

Comment:
Nit pick: Maybe save the function from wrapAndMeasure in a separate variable (e.g. `measuredDiscovery`). It will make the code a bit more readable